cmake/TaichiCXXFlags.cmake

```diff
@@ -30,7 +30,9 @@ if (WIN32)
         set(CMAKE_CXX_FLAGS
             "${CMAKE_CXX_FLAGS} /Zc:__cplusplus /std:c++17 /bigobj /wd4244 /wd4267 /nologo /Zi /D \"_CRT_SECURE_NO_WARNINGS\" /D \"_ENABLE_EXTENDED_ALIGNED_STORAGE\"")
     else()
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -g -gcodeview -fsized-deallocation -target x86_64-pc-windows-msvc")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fsized-deallocation -target x86_64-pc-windows-msvc")
```

Comment:
nits: could you use `target_compile_options`? https://cmake.org/cmake/help/latest/command/target_compile_options.html

Reply:
At this point the TaichiCore targets have not been defined (as this is the CXXFlags cmake file), also this aligns with the flags behavior on other platforms.